jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java

```diff
@@ -223,32 +243,49 @@ public Response call(String httpMethod, URL url, Request request) throws IOExcep
 
       try {
         logInsecureHttpsFailover(url);
-        return call(httpMethod, url, request, getHttpTransport(false));
+        Response response = call(httpMethod, url, request, getHttpTransport(false));
+        failoverHistory.put(url.getHost() + ":" + url.getPort(), Failover.INSECURE_HTTPS);
```

Comment:
Is this safe? Could we potenially get something like `"hello.com:8000" + "8000"` vs `"hello.com" + "8000:8000"`?

Reply:
It looks like `getHost()` excludes the port number.

Reply:
I think what I'm really asking is there anyway in which this addition can result in the same string from two different values (yes technically, but no in this case?)?

Reply:
I don't believe it can ever happen. The host portion from `URL` cannot have `:`. And the port is an integer value.

Although there can be a case where `https://example.com` and `https://example.com:443` will end up two different keys (the former goes into `https://example.com:-1`, but I intended to leave them separate. In practice the number of actual hosts involved in Jib would be merely 2 to 3 at most especially if a registry is insecure, and this keeps the code simple.